be/src/vec/exec/format/parquet/vparquet_group_reader.cpp

```diff
@@ -434,6 +438,121 @@ Status RowGroupReader::_read_empty_batch(size_t batch_size, size_t* read_rows, b
     return Status::OK();
 }
 
+Status RowGroupReader::_build_pos_delete_filter(size_t read_rows) {
+    _filter_ptr.reset(new IColumn::Filter(read_rows, 1));
+    _total_read_rows += read_rows;
+    if (!_position_delete_ctx.has_filter) {
+        return Status::OK();
+    }
+    size_t remaining_read_rows = _total_read_rows + read_rows;
+    while (_position_delete_ctx.index < _position_delete_ctx.end_index) {
+        const int64_t delete_row_index_in_row_group =
+                _position_delete_ctx.delete_rows[_position_delete_ctx.index] -
+                _position_delete_ctx.first_row_id;
+        int64_t read_range_rows = 0;
+        for (auto& range : _read_ranges) {
```

Comment:
The two-level loop is time-consuming.

Reply:
Don't search next range when there is no remaining_read_rows.